Source/ARTRealtimePresence.m

```diff
@@ -149,20 +153,54 @@ @interface ARTRealtimePresenceInternal ()
 
 NS_ASSUME_NONNULL_END
 
+typedef NS_ENUM(NSUInteger, ARTPresenceSyncState) {
+    ARTPresenceSyncInitialized,
+    ARTPresenceSyncStarted, //ItemType: nil
+    ARTPresenceSyncEnded, //ItemType: NSArray<ARTPresenceMessage *>*
+    ARTPresenceSyncFailed //ItemType: ARTErrorInfo*
+};
+
+@interface ARTEvent (PresenceSyncState)
+
+- (instancetype)initWithPresenceSyncState:(ARTPresenceSyncState)value;
++ (instancetype)newWithPresenceSyncState:(ARTPresenceSyncState)value;
+
+@end
+
 @implementation ARTRealtimePresenceInternal {
     __weak ARTRealtimeChannelInternal *_channel; // weak because channel owns self
+    __weak ARTRealtimeInternal *_realtime;
     dispatch_queue_t _userQueue;
     NSMutableArray<ARTQueuedMessage *> *_pendingPresence;
+    ARTEventEmitter<ARTEvent *, ARTPresenceMessage *> *_eventEmitter;
+    ARTDataEncoder *_dataEncoder;
+    
+    int64_t _syncMsgSerial;
+    NSString *_syncChannelSerial;
```

Comment:
```suggestion
```
Not used anywhere, so should be removed. 
`_syncChannelSerial` is only used for `sync` method moved to test utilities, so can be removed safely.

Reply:
https://github.com/ably/ably-cocoa/pull/1870/commits/1d2bec63b7bd25a6c9fbf890b1b8e0512711b686